dupserver.go

```diff
@@ -0,0 +1,56 @@
+package modgearman
+
+import (
+	"github.com/appscode/g2/client"
+)
+
+var dupjobsToSendPerServer = make(map[string](chan *answer))
+
+func initialiseDupServerConsumers(config *configurationStruct) {
+	if len(config.dupserver) > 0 {
+		for _, dupAddress := range config.dupserver {
+			logger.Debugf("creating dupserverConsumer for: %s", dupAddress)
+			dupjobsToSendPerServer[dupAddress] = make(chan *answer, config.dupServerBacklogQueueSize)
+			go runDupServerConsumer(dupAddress, dupjobsToSendPerServer[dupAddress], config)
+		}
+	}
+}
+
+func runDupServerConsumer(dupAddress string, channel chan *answer, config *configurationStruct) {
+	for {
+		item := <-channel
+		var error = sendResultDup(item, dupAddress, config)
+		if error != nil {
+			logger.Debugf("failed to send back result (to dupserver): %s", error.Error())
+			continue
+		}
+	}
+}
+
+func sendResultDup(item *answer, dupAddress string, config *configurationStruct) error {
+	var err error
+	var client *client.Client
+
+	if config.dupResultsArePassive {
+		item.active = "passive"
+	}
+
+	client, err = sendAnswer(client, item, dupAddress, config.encryption)
+
+	if client != nil {
```

Comment:
previously the client had been reused unless there were errors. I'd say it should be kept that way, otherwise the would be a new connection for each message which is a performance killer.